### PR TITLE
Improve CI & dockerfile

### PR DESCRIPTION
**Pull request by tarampampam**
This PR includes:

- `requirements.txt` file for keeping dependencies versions in the actual state using dependabot
- Common `.gitignore` file
- Additional entries in the `.dockerignore` file
- `.editorconfig` (how did you forget about this?)
- `tests` workflow (with the common steps)
- `Dockerfile` improvements
- Multi-arch docker image (`linux/amd64,linux/386,linux/arm64,linux/arm/v6,linux/arm/v7`)
- Additional Major/Major+minor docker image tags (when you publish release `1.7.3` - it will updates the `1`, `1.7`, and `1.7.3` docker tags)
- `pip` cache inside the docker image removal

Have a nice day!

### CODE REVIEW

**Comment by tarampampam** [comment]
Friendly ping @jetexe

**Comment by tarampampam** [comment]
As you can see [here](https://github.com/tarampampam/cookiecutter-docker/pull/1) - dependabot works fine for now!

**Comment by tarampampam** [comment]
And one more important thing - after this PR merging your releasing process will be changed from "manually checking for the cookie-cutter releases and when a new release has been published - make a release in your project with the same version" to "when dependabot makes a PR with cookie-cutter update - make a release with the cookie-cutter version". Build arguments is no longer needed

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -2,8 +2,7 @@ name: release
 
 on:
   release:
-    types: # This configuration does not affect the page_build event above
-      - created
+    types: [published]
```
Comment (tarampampam) [inline .github/workflows/release.yml:5]: Only for the published releases (not for the **pre**-releases)

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -12,6 +11,9 @@ jobs:
       - name: Check Out Repo
         uses: actions/checkout@v2
 
+      - uses: gacts/github-slug@v1 # Action page: <https://github.com/gacts/github-slug>
+        id: slug
```
Comment (tarampampam) [inline .github/workflows/release.yml:15]: This is my own GitHub action for the version extracting. Forget about `echo ${GITHUB_REF##*/} | sed -e 's/^[vV ]*//'` forever 😄

**Comment by jetexe** [comment]
You've done a tremendous job! Thank you

**Pull request merged**